Improve legend for staging states

eduardoj opened the pull request:
Fixes #15524.

### Before

| Light mode | Dark mode |
| --- | --- |
| ![Screenshot From 2025-05-20 10-39-29](https://github.com/user-attachments/assets/ad1faf06-28cd-4909-b607-17d4ab36e00d) | ![Screenshot From 2025-05-20 10-39-39](https://github.com/user-attachments/assets/e59647fa-656d-42a4-abae-a89e205b8666) |


### After

| Light mode | Dark mode |
| --- | --- |
| ![Screenshot From 2025-05-20 12-39-46](https://github.com/user-attachments/assets/5d517cfb-f312-4f77-a4bd-4060064fcd2e) | ![Screenshot From 2025-05-20 12-39-51](https://github.com/user-attachments/assets/110c5882-c6bb-4b11-be12-f7678bae1c7a) |

hellcp-work commented:
I feel like this legend is harder to read, and obsolete still doesn't stand out from the rest of the types

eduardoj commented:
> I feel like this legend is harder to read,

Why exactly?

> and obsolete still doesn't stand out from the rest of the types

Yeah, agree. This deserves a follow-up.

hellcp-work commented:
Rectangles with colour in them is how a legend is usually done, so it was easy to make that mental connection. Putting the entire thing in a badge makes this a harder association. Also the delete really shouldn't have a background at all

eduardoj commented:
> Rectangles with colour in them is how a legend is usually done, so it was easy to make that mental connection.

I see. I tried to come up with a legend which didn't show the colors in rectangles (with black or white borders depending on the light/dark mode). Instead, I wanted to show exactly how an element would look like. I find this option more intuitive.

> Putting the entire thing in a badge makes this a harder association.

It is how it is shown in the end. The association here is reduced to the minimum. What you see is what you get.

> Also the delete really shouldn't have a background at all

But that is not how things are shown in the end. That would deviate from the final rendering. What do you suggest to use as background? A transparent background?

hellcp-work commented:
> But that is not how things are shown in the end. That would deviate from the final rendering. What do you suggest to use as background? A transparent background?

Yes, as delete can be applied to any other color

eduardoj commented:
> > But that is not how things are shown in the end. That would deviate from the final rendering. What do you suggest to use as background? A transparent background?
> 
> Yes, as delete can be applied to any other color

What is your suggestion for dark mode?

eduardoj commented:
> > and obsolete still doesn't stand out from the rest of the types
> 
> Yeah, agree. This deserves a follow-up.

I added a commit with a new color theme: 'custom-gray'. I updated the images in the description of the PR. Please take a look again.
